lib/pure/streams.nim

```diff
@@ -53,12 +53,17 @@ type
       {.nimcall, raises: [Defect, IOError, OSError], tags: [], gcsafe.}
     getPositionImpl*: proc (s: Stream): int
       {.nimcall, raises: [Defect, IOError, OSError], tags: [], gcsafe.}
+
+    readDataStrImpl*: proc (s: Stream, buffer: var string, slice: Slice[int]): int
```

Comment:
Would it be possible, instead of adding another field, pass an openArray to readData?

Reply:
I wish, but I had no luck with ``openArray`` slices of a string at compile time. I guess ``proc (s: Stream, buffer: var openarray[char], slice: Slice[int]): int`` could work, but I doubt that it would be much better.